packages/nimble/R/MCMC_conjugacy.R

```diff
@@ -495,34 +495,40 @@ conjugacyClass <- setRefClass(
-                ##                           ))
-                ## neededParams <- dependents[[distName]]$neededParamsForPosterior
-                ## for(param in neededParams) {
-                ##     depNodeParamNdim <- getDimension(distName, param) 
-                ##     ## NEWNODEFXN
-                ##     functionBody$addCode(DEP_PARAM_VAR <- array(0, dim = DECLARE_SIZE),
-                ##                          list(DEP_PARAM_VAR      = as.name(paste0('dep_', distName, '_', param)),              ## DECLARE() statement
-                ##                               DECLARE_SIZE       = makeDeclareSizeField(as.name(paste0('N_dep_',distName)), depNodeParamNdim)))   ## won't run since adding another argument to makeDeclareSizeField
-                ## }
-
-                ## functionBody$addCode({
-                ##     N_DEP <- length(control$DEP_CONTROL_NAME)
-                ##     DEP_NODEFUNCTIONS <- nimbleFunctionList(NF_VIRTUAL)
-                ##     for(iDep in 1:N_DEP)
-                ##         DEP_NODEFUNCTIONS[[iDep]] <- model$nodes[[control$DEP_CONTROL_NAME[iDep]]]
-                ## }, list(N_DEP             = as.name(paste0('N_dep_', distName)),
-                ##         DEP_CONTROL_NAME  = as.name(paste0(  'dep_', distName)),
-                ##         DEP_NODEFUNCTIONS = as.name(paste0(  'dep_', distName, '_nfs')),
-                ##         NF_VIRTUAL        = as.name(paste0('node_stoch_', distName))))
+                
+                ## declare() statements are removed from run() code,
+                ## and were replaced with setup output array() calls below.
+                ## July 2017
+                depNodeValueNdim <- getDimension(distName)
+                functionBody$addCode(DEP_VALUES_VAR <- array(0, dim = DECLARE_SIZE),
+                                     list(DEP_VALUES_VAR = as.name(paste0('dep_', distName, '_values')),
+                                          DECLARE_SIZE   = makeDeclareSizeField(as.name(paste0('N_dep_', distName)), as.name(paste0('dep_', distName, '_nodeSizeMax')), as.name(paste0('dep_', distName, '_nodeSizeMax')), depNodeValueNdim)))
+                neededParams <- dependents[[distName]]$neededParamsForPosterior
+                for(param in neededParams) {
+                    depNodeParamNdim <- getDimension(distName, param)
+                    functionBody$addCode(DEP_PARAM_VAR <- array(0, dim = DECLARE_SIZE),
+                                         list(DEP_PARAM_VAR = as.name(paste0('dep_', distName, '_', param)),
+                                              DECLARE_SIZE  = makeDeclareSizeField(as.name(paste0('N_dep_', distName)), as.name(paste0('dep_', distName, '_nodeSizeMax')), as.name(paste0('dep_', distName, '_nodeSizeMax')), depNodeParamNdim)))
+                }
             }
-
+            
+            ## more new array() setup outputs, instead of declare() statements, for offset and coeff variables
+            ## July 2017
```

Comment:
Rational for `git blame`: Consider two strategies: manually add historical comments to a select section of code, or rely entirely on automatic history annotation. While the latter strategy is less customizable, it has the major advantage that *every line of code* has history annotations. And while there is initial investment cost in gaining proficiency with `git blame` and `git grep`, this quickly pays off in saved time for code writers, and more focused attention for code readers (who more often than not are uninterested in history).